sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -87,23 +105,77 @@ def get_base_images_sbom_components(base_images_digests, is_last_from_scratch):
     return components
 
 
+def get_base_images_from_dockerfile(parsed_dockerfile):
+    """
+    Reads the base images from provided parsed dockerfile
+
+    :param parsed_dockerfile: (Dict) - Contents of the parsed dockerfile
+    :return: base_images (List) - List of base images used during build as extracted
+                                  from the dockerfile in the order they were used.
+
+    Example:
+    If the Dockerfile looks like
+    FROM registry.access.redhat.com/ubi8/ubi:latest as builder
+    ...
+    FROM builder
+    ...
+
+    Then the relevant part of parsed_dockerfile look like
+    {
+        "Stages": [
+            {
+                "BaseName": "registry.access.redhat.com/ubi8/ubi:latest",
+                "As": "builder",
+                "From": {"Image": "registry.access.redhat.com/ubi8/ubi:latest"},
+            },
+            {
+                "BaseName": "builder",
+                "From": {"Stage": {"Named": "builder", "Index": 0}},
+            },
+        ]
+    },
+    """
+    base_images = []
+
+    # this part of the json is the relevant one that contains the
+    # info about base images
+    stages = parsed_dockerfile["Stages"]
+
+    for stage in stages:
+        if "Image" in stage["From"]:
+            base_images.append(stage["From"]["Image"])
+        elif "Scratch" in stage["From"]:
+            base_images.append("scratch")
+        elif "Stage" in stage["From"]:
+            stage_refering_to_index = stage["From"]["Stage"]["Index"]
```

Comment:
nitpick: I thing `stage_index` would be clearer. `stage_refering_to_index` sounds like the referenced Stage object, not the index

Reply:
I really wanted to convey that is is not the stage of the current index, but that it is a reference to another stage. But changed it to `stage_index` and the rest is clearer from context.